netutils/config/compliance.py

```diff
@@ -27,7 +29,7 @@
 }
 
 
-def _check_configs_differences(intended_cfg, actual_cfg, network_os):
+def _check_configs_differences(intended_cfg: str, actual_cfg: str, network_os: str) -> t.Dict[str, t.Union[str, bool]]:
```

Comment:
If I see this correctly this would be another place we could use the proposed `default_feature` `TypedDict`.